### PR TITLE
[KERNEL-ESTIMATOR] [ObjectivityLtd] [31/10/2022]

### PR DESCRIPTION
**Pull request by mbaczyk123**
### Team members
Michal Baczyk
Peter den Haan

### CODE REVIEW

**Comment by ryanhill1** [comment]
@mbaczyk123 @drpdh what is the email address associated with the qBraid account I should look under?

**Comment by mbaczyk123** [comment]
> @mbaczyk123 @drpdh what is the email address associated with the qBraid account I should look under?

Apologies for not including it sooner.
mbaczyk@objectivity.co.uk

**Review by ryanhill1 — COMMENTED** [review]
Thanks for your submission! From what I can see, the way you've calculated `kernel_experimental()` looks good. However, it doesn't seem like you've fully followed step 17 of "Algorithm 1: training":

> 17. Use quadratic program solver to optimize $\alpha$ in $L_D$ in eqn. (1) with kernel $K = \hat{K}$ and set $T$.

You use `SVC.fit`, but don't seem to have defined the function $L_D$ that should be used to calculate the loss when doing the fit. Right now, it seems you are fitting the model using whatever the default loss function is for the `SVC.fit` method, which wouldn't follow the algorithm.

Additionally, in your `classifier()` function, it seems like you've dropped the $y_i$ term in

$\tilde{m}(\textbf{s}) = \text{sign} \bigg(\sum{_{i=1}^t} y_i \alpha_i^* K(\textbf{x}_i, \textbf{s}) + b \bigg)$

Let me know if I've misinterpreted anything, or if you have any questions!
